be/src/exec/olap_scanner.cpp

```diff
@@ -152,14 +152,15 @@ Status OlapScanner::_init_tablet_reader_params(
                       ->rowset()
                       ->rowset_meta()
                       ->is_segments_overlapping());
+    
+    bool has_replace_col = false;
+    RETURN_IF_ERROR(_init_return_columns(!single_non_overlapping_version, has_replace_col));
 
-    _tablet_reader_params.direct_mode = single_version || _aggregation;
```

Comment:
The issue here is:
1. Planner thinks scanner can scan directly, don't need merge and can do some pre-aggregation works.
2. But scanner found it's a unqiue key table with seq column which need to merge first.

I think we'd fix it in planner? Planner should not set preAggregation as true on a unique table with seq column?

Reply:
I am not sure what we need to in planner.  If fe sets aggregation, we ignore it here, then we can ignore aggreagate in planner somtimes?  Could you throw an example sql?